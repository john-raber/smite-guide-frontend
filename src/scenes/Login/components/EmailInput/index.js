import React from 'react';
import { FormGroup, Label, Input } from 'reactstrap';

const EmailInput = ({ value, onChange }) => {
  return (
    <FormGroup className="EmailInput">
      <Label for="email">Email</Label>
      <Input
        className="login-email"
        name="email"
        type="email"
        value={value}
        onChange={e => onChange(e.target.name, e.target.value)}
      />
    </FormGroup>
  );
};

export default EmailInput;
