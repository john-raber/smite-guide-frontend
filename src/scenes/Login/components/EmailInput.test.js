import React from "react";
import { shallow } from "enzyme";

import EmailInput from "./EmailInput/index";

describe("EmailInput", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<EmailInput />);

    expect(wrapper.find(".EmailInput").length).toEqual(1);
  });

  it("matches its snapshot", () => {
    let wrapper = shallow(<EmailInput />);

    expect(wrapper).toMatchSnapshot();
  });
});
