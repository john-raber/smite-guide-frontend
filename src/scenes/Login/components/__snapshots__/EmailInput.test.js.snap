// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput matches its snapshot 1`] = `
<FormGroup
  className="EmailInput"
  tag="div"
>
  <Label
    for="email"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Email
  </Label>
  <Input
    name="email"
    onChange={[Function]}
    type="email"
  />
</FormGroup>
`;
